### PR TITLE
Clean up tests

### PR DESCRIPTION
**Pull request by franmomu**
Similar to https://github.com/sonata-project/SonataAdminBundle/pull/5794

I'm targeting this branch because this is BC.

### CODE REVIEW

File: tests/Filesystem/LocalTest.php [hunk 1]
```diff
@@ -23,7 +23,7 @@ public function testReplicate(): void
         $local = new Local('/tmp');
 
         // check if OS is Mac OS X where /tmp is a symlink to /private/tmp
-        $result = 'Darwin' === php_uname('s') ? '/private/tmp' : '/tmp';
+        $result = 'Darwin' === PHP_OS ? '/private/tmp' : '/tmp';
```
Comment (greg0ire) [inline tests/Filesystem/LocalTest.php]: I wonder if this could be simplified by using `sys_get_tmp_dir()`

File: .travis.yml [hunk 2]
```diff
@@ -36,7 +36,7 @@ env:
     - SYMFONY_DEPRECATIONS_HELPER=max[self]=0
     - TARGET=test
     - UPSTREAM_URL=https://github.com/sonata-project/SonataMediaBundle.git
-    - PHPUNIT_VERSION=7
+    - PHPUNIT_VERSION=8
```
Comment (greg0ire) [inline .travis.yml]: You will have to make a PR on dev-kit to change this

File: tests/Filesystem/LocalTest.php (hunk 1 above)
Comment (franmomu) [reply tests/Filesystem/LocalTest.php]: mmm using `sys_get_temp_dir` with Mac OS X I get something like `/var/folders/48/fkmnn6gs2998gq8wc5f83d780000gn/T`

Comment (greg0ire) [reply tests/Filesystem/LocalTest.php]: Ok, doesn't look like it…

**Comment by greg0ire** [comment]
To be merged in sync with the dev-kit PR

**Comment by OskarStark** [comment]
Can we merge this?

File: tests/Block/FeatureMediaBlockServiceTest.php [hunk 3]
```diff
@@ -32,7 +32,7 @@ protected function setUp(): void
         $this->galleryManager = $this->prophesize(GalleryManagerInterface::class);
 
         $this->blockService = new FeatureMediaBlockService(
-            'block.service',
+            $this->templating,
```
Comment (OskarStark) [inline tests/Block/FeatureMediaBlockServiceTest.php]: templating twice?

File: tests/Block/GalleryBlockServiceTest.php [hunk 4]
```diff
@@ -35,7 +47,7 @@ protected function setUp(): void
         $this->galleryManager = $this->prophesize(GalleryManagerInterface::class);
 
         $this->blockService = new GalleryBlockService(
-            'block.service',
+            $this->templating,
```
Comment (OskarStark) [inline tests/Block/GalleryBlockServiceTest.php]: Same here

File: tests/Admin/BaseMediaAdminTest.php [hunk 5]
```diff
@@ -29,10 +29,29 @@
 
 class BaseMediaAdminTest extends TestCase
 {
+    /**
+     * @var ObjectProphecy|Pool
```
Comment (core23) [inline tests/Admin/BaseMediaAdminTest.php]:
```suggestion
     * @var ObjectProphecy&Pool
```

The mock will provide both methods

File: tests/Block/FeatureMediaBlockServiceTest.php (hunk 3 above)
Comment (franmomu) [reply tests/Block/FeatureMediaBlockServiceTest.php]: The thing is that [`MediaBlockService` expects a `name` and a `EngineInterface`](https://github.com/sonata-project/SonataMediaBundle/blob/3.x/src/Block/MediaBlockService.php#L56) and [passing a `name` as string is deprecated](https://github.com/sonata-project/SonataBlockBundle/blob/3.x/src/Block/Service/AbstractBlockService.php#L51)

Comment (OskarStark) [reply tests/Block/FeatureMediaBlockServiceTest.php]: Ok!

**Comment by greg0ire** [comment]
Thanks @franmomu

**Pull request merged**